repository/src/main/java/org/alfresco/opencmis/AlfrescoCmisServiceFactory.java

```diff
@@ -217,6 +230,26 @@ public CmisService getService(final CallContext context)
     
     protected AlfrescoCmisService getCmisServiceTarget(CMISConnector connector)
     {
-        return new AlfrescoCmisServiceImpl(connector);
+        AlfrescoCmisServiceImpl cmisService = new AlfrescoCmisServiceImpl(connector);
+        cmisService.setCmisRequestRenditionsOnCreateDoc(parseCommaSeparatedSet(getCmisCreateDocRequestRenditionsSet()));
+        return cmisService;
+    }
+
+    private Set<String> parseCommaSeparatedSet(String str)
+    {
+        Set<String> stringSet = new HashSet<>();
+        if (str != null)
+        {
+            StringTokenizer st = new StringTokenizer(str, ",");
+            while (st.hasMoreTokens())
+            {
+                String entry = st.nextToken().trim();
+                if (!entry.isEmpty())
+                {
+                    stringSet.add(entry);
+                }
+            }
+        }
+        return stringSet;
     }

```

Comment:
Looks good to me, is it possible to add a unit test case for this?

Reply:
I didn't notice any existing CMIS tests for checking *async* renditions. 

@nithinnambiar How would you propose to unit test that option is passed in ok (when not empty) ? Note: I have checked manually via debug. 

Instead, I suggest we add a log info message allowing easy verification both ways. 

Does that sound reasonable for an optional setting (and no other change in functionality) ?

Reply:
Perhaps a small unit test to check if  `cmisRequestRenditionsOnCreateDoc`  in `AlfrescoCmisServiceImpl` is getting populated with right values from the config?

Reply:
Seems overkill. Do you have examples of a similar test elsewhere in the codebase - if so, happy to add ;-)